File: backend/DataAccess/Common/Models/DaFileModel.cs
// *******************************************************************
// 業務名称　: 健康管理システム
// 機能概要　: ファイルモデル
//
// 作成日　　: 2023.02.14
// 作成者　　: 蔣
// 変更履歴　:
// *******************************************************************
using Newtonsoft.Json;

namespace BCC.Affect.DataAccess
{
    public class DaFileModel
    {
        public string filenm { get; set; } = string.Empty;             //ファイル名
        public string filetype { get; set; } = string.Empty;           //ファイルタイプ
        public int filesize { get; set; } = 0;                         //ファイルサイズ
        [JsonIgnore]
        public byte[] filedata { get; set; } = Array.Empty<byte>();    //ファイルデータ
        public DaFileModel(string filenm, string filetype, byte[] filedata)
        {
            this.filenm = filenm;
            this.filetype = filetype;
            this.filedata = filedata;
            this.filesize = filedata.Length;
        }
    }
}

File: backend/BusinessService/Db/Common/Models/DaFileModel.vb
' *******************************************************************
' 業務名称　: 互助防疫システム
' 機能概要　: ファイルモデル
'
' 作成日　　: 2024.07.14
' 作成者　　: 
' 変更履歴　:
' *******************************************************************
Imports System.Text.Json.Serialization
Imports Newtonsoft.Json

Namespace JBD.GJS.Db
    Public Class DaFileModel
        Public Property filenm As String = String.Empty             'ファイル名
        Public Property filetype As String = String.Empty           'ファイルタイプ
        Public Property filesize As Integer = 0                         'ファイルサイズ
        <JsonIgnore>
        Public Property filedata As Byte() = Array.Empty(Of Byte)()    'ファイルデータ
        Public Sub New(filenm As String, filetype As String, filedata As Byte())
            Me.filenm = filenm
            Me.filetype = filetype
            Me.filedata = filedata
            filesize = filedata.Length
        End Sub
    End Class
End Namespace
